PR: https://github.com/odoo/odoo/pull/236918

From: ecc8537fe2f29209e2672f54cc0eca01a9e7039a
From: chdh-odoo
Date: 2026-04-16 04:01:03

Breaking data model changes scores: del:8 + add:0, change matches:
-    social_twitter = fields.Char('X Account', default=_default_social_twitter)
-    social_facebook = fields.Char('Facebook Account', default=_default_social_facebook)
-    social_github = fields.Char('GitHub Account', default=_default_social_github)
-    social_linkedin = fields.Char('LinkedIn Account', default=_default_social_linkedin)
-    social_youtube = fields.Char('Youtube Account', default=_default_social_youtube)
-    social_instagram = fields.Char('Instagram Account', default=_default_social_instagram)
-    social_tiktok = fields.Char('TikTok Account', default=_default_social_tiktok)
-    social_discord = fields.Char('Discord Account', default=_default_social_discord)

Total Changes: 32

[IMP] website, *: social media snippet improvements

*=html_builder, website_sale, website_blog

Purpose:
* This commit makes social media links behave independently instead of
  syncing across snippets.
* Previously, editing social links in one snippet affected all other
  snippets, which was confusing.
* We no longer introduce new social routes for social snippets.
  Using the actual social media links is preferable as it avoids
  redirects, resolves directly in the HTML, and better SEO.
* We now prefill all social media links using company social media
  links, if no company social media links present in company then
  set demo link.
* The sidebar is simplified by removing the toggle link option and
  adding a Replace Icon buton.
* When no valid icon matches a link, a default pencil icon is used.
* Restrict media dialog to icons and images tab only.
* Remove social route and fields from website model since it will be
  useless.

task-5346732

closes odoo/odoo#236918

Related: odoo/upgrade#9593
Signed-off-by: Sébastien Blondiau (blse) <blse@odoo.com>

================================= pseudo patch: =================================

--- a/addons/website/models/website.py
+++ b/addons/website/models/website.py
@@ -164,43 +164,11 @@ class Website(models.CachedModel):
         'List of blocked 3rd-party domains',
         compute='_compute_blocked_third_party_domains')
 
-    def _default_social_facebook(self):
-        return self.env.ref('base.main_company').social_facebook
-
-    def _default_social_github(self):
-        return self.env.ref('base.main_company').social_github
-
-    def _default_social_linkedin(self):
-        return self.env.ref('base.main_company').social_linkedin
-
-    def _default_social_youtube(self):
-        return self.env.ref('base.main_company').social_youtube
-
-    def _default_social_instagram(self):
-        return self.env.ref('base.main_company').social_instagram
-
-    def _default_social_twitter(self):
-        return self.env.ref('base.main_company').social_twitter
-
-    def _default_social_tiktok(self):
-        return self.env.ref('base.main_company').social_tiktok
-
-    def _default_social_discord(self):
-        return self.env.ref('base.main_company').social_discord
-
     def _default_logo(self):
         with file_open('website/static/src/img/website_logo.svg', 'rb') as f:
             return BinaryBytes(f.read())
 
     logo = fields.Binary('Website Logo', default=_default_logo, help="Display this logo on the website.")
-    social_twitter = fields.Char('X Account', default=_default_social_twitter)
-    social_facebook = fields.Char('Facebook Account', default=_default_social_facebook)
-    social_github = fields.Char('GitHub Account', default=_default_social_github)
-    social_linkedin = fields.Char('LinkedIn Account', default=_default_social_linkedin)
-    social_youtube = fields.Char('Youtube Account', default=_default_social_youtube)
-    social_instagram = fields.Char('Instagram Account', default=_default_social_instagram)
-    social_tiktok = fields.Char('TikTok Account', default=_default_social_tiktok)
-    social_discord = fields.Char('Discord Account', default=_default_social_discord)
     social_default_image = fields.Binary(string="Default Social Share Image", help="If set, replaces the website logo as the default social share image.")
     has_social_default_image = fields.Boolean(compute='_compute_has_social_default_image', store=True)
 
